### PR TITLE
Add jest config to `blitz-next` package

### PR DESCRIPTION
**Pull request by beerose**
Closes: https://github.com/blitz-js/blitz/issues/3216

## Bug Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)

## Feature Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)
- [ ] Documentation added/updated (submit PR to [blitzjs.com repo `canary` branch](https://github.com/blitz-js/blitzjs.com/tree/canary))

### CODE REVIEW

File: apps/web/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
   "extends": "@blitzjs/config/tsconfig.nextjs.json",
+  "compilerOptions": {
+    // "isolatedModules": false
+  },
```
Comment (beerose) [inline apps/web/tsconfig.json:5]:
```suggestion
```

File: apps/web/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+const createJestConfig = nextJest({
+  dir: "./",
+})
+
+const customJestConfig = {
+  testEnvironment: "jest-environment-jsdom",
+}
+
+module.exports = createJestConfig(customJestConfig)
```
Comment (beerose) [inline apps/web/jest.config.js:11]: I assumed we want to have the same API ad `next/jest`: https://nextjs.org/docs/testing

File: packages/blitz-next/jest/client/setup-after-env.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+require("@testing-library/jest-dom")
+process.env.__BLITZ_SESSION_COOKIE_PREFIX = "blitz"
```
Comment (flybayer) [inline packages/blitz-next/jest/client/setup-after-env.js:2]: remove line

Comment (beerose) [inline packages/blitz-next/jest/client/setup-after-env.js:2]:
```suggestion
```

**Pull request merged**